pkg/api/api.go

```diff
@@ -100,13 +100,14 @@ type AccessService interface {
 
 // AccessRuleService can create and get rules
 type AccessRuleService interface {
+	ArchiveAccessRule(ctx context.Context, userID string, in rule.AccessRule) (*rule.AccessRule, error)
+	CreateAccessRule(ctx context.Context, userID string, in types.CreateAccessRuleRequest) (*rule.AccessRule, error)
 	LookupRule(ctx context.Context, opts rulesvc.LookupRuleOpts) ([]rulesvc.LookedUpRule, error)
-	ArchiveAccessRule(ctx context.Context, user *identity.User, in rule.AccessRule) (*rule.AccessRule, error)
-	CreateAccessRule(ctx context.Context, user *identity.User, in types.CreateAccessRuleRequest) (*rule.AccessRule, error)
 	GetRule(ctx context.Context, ID string, user *identity.User, isAdmin bool) (*rule.GetAccessRuleResponse, error)
 	UpdateRule(ctx context.Context, in *rulesvc.UpdateOpts) (*rule.AccessRule, error)
 	RequestArguments(ctx context.Context, accessRuleTarget rule.Target) (map[string]types.RequestArgument, error)
 }
+
 type CacheService interface {
 	RefreshCachedProviderArgOptions(ctx context.Context, providerId string, argId string) (bool, []cache.ProviderOption, []cache.ProviderArgGroupOption, error)
 	LoadCachedProviderArgOptions(ctx context.Context, providerId string, argId string) (bool, []cache.ProviderOption, []cache.ProviderArgGroupOption, error)
```

Comment:
Nitpick here (not sure if this is related to this specific PR) - what's the difference between `RefreshCachedProviderArgOptions` and `LoadCachedProviderArgOptions`? In the `validateTargetArgumentAgainstCachedOptions` method in this PR, we've swapped from one method to another, but I find it difficult to understand the difference between these two methods in a code review.

My assumption is that `LoadCachedProviderArgOptions` simply returns cached values, whereas `RefreshCachedProviderArgOptions` invalidates the cache.

Could we refactor these to be:

```go
type LoadArgOptionsOpts struct {
  ProviderID string
  ArgID string
  // RefreshCache invalidates the cache and refreshes argument options using the Access Provider.
  RefreshCache bool
}

type CacheService interface {
    LoadArgOptions(ctx context.Context, opts LoadArgOptionsOpts) (bool, []cache.ProviderOption, []cache.ProviderArgGroupOption, error)
}
```

Reply:
My concern is that the existing implementation makes it very easy to mix up `RefreshCachedProviderArgOptions` with `LoadCachedProviderArgOptions`

Reply:
I see where you are coming from here @chrnorm. I can make the change, but I think this will need to be refactored anyway with the inclusion of the new v2 providers provider arg options, this is what I rewrote today actually. 

I suggest we make a ticket to remind to update this when that is all merged in. (I'd say this will be a week or so)

Reply:
Sounds good - will add this to our backlog